tools/cli/domainCommands.go

```diff
@@ -288,6 +288,16 @@ func (d *domainCLIImpl) DeprecateDomain(c *cli.Context) {
 	ctx, cancel := newContext(c)
 	defer cancel()
 
+	// check if there is any workflow in this domain, if exists, do not deprecate
+	wfs, _ := listClosedWorkflow(getWorkflowClient(c), 1, 0, time.Now().UnixNano(), "", "", workflowStatusNotSet, nil, c)
+	if len(wfs) > 0 {
+		ErrorAndExit("Operation DeprecateDomain failed.", errors.New("Workflow history not cleared in this domain."))
```

Comment:
why closed workflows still matter? Is that a requirements or some technical workaround?

Reply:
it's a requirement. We want to be conservative.

Reply:
May I ask what's purpose of this change? I thought only start or signal workflow is blocked for deprecated domains. Trying to understand why do closed workflow and running workflow matter here?

If we want to enforce this requirement that domain deprecation only happens for domains in which all workflows have passed retention, I think this check should be done inside the handler of domain deprecation call instead of just in the CLI?

Reply:
The reason I made the change in CLI but not in server is to let the caller  do the check.
And within CLI, we can add another flag (e.g. --force) to make this check optional, but the default behavior is to be conservative and does the check.

Reply:
@meiliang86 What's your opinion on this? I prefer to adding the check in CLI and later we can add a flag to skip the check if we'd like to.

Reply:
Yeah adding the check to the CLI is fine. Technically we can deprecate domains even there's open/close workflows, so it's ok to not check it on the server side.
This extra check is just to make sure we don't shoot ourselves in the foot (i.e. accidentally deprecating domains that are still in use), as right now we don't support resurrecting domains once deprecated.